### PR TITLE
Review Integrations for StateManager Cache PR #2630

### PR DESCRIPTION
**Pull request by holgerd77**
Review integrations for #2630, confirmation comments are done on the original review comments.

### CODE REVIEW

**Comment by holgerd77** [comment]
Updated this branch to latest `develop-v7` rebase.

**Comment by holgerd77** [comment]
@acolytec3 can this be merged then (would need re-approval from you)?

File: packages/evm/src/evm.ts [hunk 1]
```diff
@@ -635,7 +635,7 @@ export class EVM implements EVMInterface {
         //await this._state.putContractCode(message.to, result.returnValue)
```
Comment (acolytec3) [inline packages/evm/src/evm.ts:635]: Can we remove this commented out code?

File: packages/client/test/rpc/eth/getBalance.spec.ts [hunk 2]
```diff
@@ -81,6 +81,14 @@ tape(`${method}: ensure balance deducts after a tx`, async (t) => {
   // call with height that exceeds chain height
   req = params(method, [address.toString(), '0x1'])
   expectRes = checkError(t, INVALID_PARAMS, 'specified block greater than current height')
+  await baseRequest(t, server, req, 200, expectRes, false)
+
+  // call with not existing account
```
Comment (acolytec3) [inline packages/client/test/rpc/eth/getBalance.spec.ts:86]:
```suggestion
  // call with nonexistent account
```

File: packages/client/test/rpc/eth/getTransactionCount.spec.ts [hunk 3]
```diff
@@ -74,6 +74,14 @@ tape(`${method}: call with valid arguments`, async (t) => {
     const msg = 'should return the correct nonce (1)'
     t.equal(res.body.result, '0x1', msg)
   }
+  await baseRequest(t, server, req, 200, expectRes, false)
+
+  // call with not existing account
```
Comment (acolytec3) [inline packages/client/test/rpc/eth/getTransactionCount.spec.ts:79]:
```suggestion
  // call with nonexistent account
```

File: packages/client/test/rpc/eth/getTransactionCount.spec.ts [hunk 4]
```diff
@@ -74,6 +74,14 @@ tape(`${method}: call with valid arguments`, async (t) => {
...
+  req = params(method, [`0x${'11'.repeat(20)}`, 'latest'])
+  expectRes = (res: any) => {
+    const msg = 'should return 0x0 for not existing account'
```
Comment (acolytec3) [inline packages/client/test/rpc/eth/getTransactionCount.spec.ts:82]:
```suggestion
    const msg = 'should return 0x0 for nonexistent account'
```

File: packages/client/test/rpc/eth/getBalance.spec.ts [hunk 5]
```diff
@@ -81,6 +81,14 @@ tape(`${method}: ensure balance deducts after a tx`, async (t) => {
...
+  req = params(method, [`0x${'11'.repeat(20)}`, 'latest'])
+  expectRes = (res: any) => {
+    const msg = 'should return 0x0 for not existing account'
```
Comment (acolytec3) [inline packages/client/test/rpc/eth/getBalance.spec.ts:89]:
```suggestion
    const msg = 'should return 0x0 for nonexistent account'
```

**Review by acolytec3 — APPROVED** [review]
Looks good.  A couple of small nits that can be changed or not.

**Comment by holgerd77** [comment]
@acolytec3 have integrated and updated the branch, should be ready for merge if tests pass!

**Review by acolytec3 — APPROVED** [review]
LGTM!

**Pull request merged**